wave_calculator.gd

```diff
@@ -12,25 +12,52 @@ func _init(enemies: Dictionary) -> void:
 
 # Array[Dictionary] doesn't work :sad: https://forum.godotengine.org/t/array-with-dictionaries-doesnt-get-recognized-as-array-dictionary/58280
 func get_enemy_spawns(wave_number: int) -> Array:
-	var difficulty = _get_difficulty(wave_number)
-	var max_batch_score = _get_max_batch_score(difficulty)
-	var total_batches = max(1, ceil(float(difficulty) / max_batch_score))
-
 	var result = []
-	var total_wave_score = 0
-	var enemy_choices = _enemies.keys().filter(_lessThanEqualTo(_get_max_enemy_score(difficulty)))
 
-	for _i in range(total_batches):
-		var target_batch_score = min(difficulty - total_wave_score, max_batch_score)
-		total_wave_score += target_batch_score
-		result.append(_batch(target_batch_score, enemy_choices))
+	var difficulty = _calculate_wave_difficulty(wave_number)
+	var max_enemy_cost = _calculate_max_enemy_cost(difficulty)
+	var enemy_choices = _enemies.keys().filter(_less_than_equal_to(max_enemy_cost))
+	var avg_enemy_cost = floor((enemy_choices.reduce(func(a, b): return a + b))/len(enemy_choices))
```

Comment:
would prefer to use the `_sum` method but i can't get GDScript to not throw at runtime due to it claiming the `enemy_choices` array (which should be Array[int]) not matching the arg type for `_sum(array: Array[int])` :cry:

Reply:
removed the avg enemy cost